transport/src/main/java/io/netty/channel/ChannelConfig.java

```diff
@@ -166,14 +167,28 @@
      * Returns {@link ByteBufAllocator} which is used for the channel
      * to allocate buffers.
      */
+    @Deprecated
     ByteBufAllocator getAllocator();
 
     /**
      * Set the {@link ByteBufAllocator} which is used for the channel
      * to allocate buffers.
      */
+    @Deprecated
     ChannelConfig setAllocator(ByteBufAllocator allocator);
 
+    /**
+     * Returns {@link BufferAllocator} which is used for the channel
+     * to allocate buffers.
+     */
+    BufferAllocator getBufferAllocator();
+
+    /**
+     * Set the {@link ByteBufAllocator} which is used for the channel
+     * to allocate buffers.
+     */
+    ChannelConfig setBufferAllocator(BufferAllocator allocator);
```

Comment:
@NiteshKant you will need to override this method in all `ChannelConfig` sub-interfaces so it returns the correct type for method chaining.

Reply:
aah yes!